src/strategies/vesper/index.ts

```diff
@@ -0,0 +1,72 @@
+import { formatUnits } from '@ethersproject/units';
+import { getBlockNumber } from '../../utils';
+import { Multicaller } from '../../utils';
+
+export const author = 'Vesper Finance';
+export const version = '1.0.0';
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const snapshotBlock =
+    typeof snapshot === 'number' ? snapshot : await getBlockNumber(provider);
+  const snapshotBlocks: number[] = [];
+
+  const { blocksPerPeriod, periods, minBlock } = options;
+
+  for (let i = 0; i < periods; i++) {
+    const blockTag = snapshotBlock - blocksPerPeriod * i;
+    if (blockTag < minBlock) {
+      break;
+    }
+    snapshotBlocks.push(blockTag);
+  }
+
+  const scores = await Promise.all([
+    ...snapshotBlocks.map((blockTag) =>
+      getScores(provider, addresses, options, blockTag)
+    )
+  ]);
```

Comment:
@marcelomorgado I think we should have a limit to `periods` param else if someone passes thousands of previous blocks, strategy won't scale 
ideally, it should be like `5` maximum blocks maybe?


The limit of the number of requests/multicall is `5` 
https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Reply:
Done. I've hardcoded `period = 4`